packages/api/src/apps/lsp_api/app.ts

```diff
@@ -220,75 +209,77 @@ export default async (env: ProcessEnv) => {
   }
 
   // wait update rate before running loops, since all state was just updated on init
-  await new Promise((res) => setTimeout(res, updateRateS * 1000));
-
-  console.log("Starting LSP API update loops");
-
-  // listen for new lsp contracts since after we have started api, and make sure they get state updated asap
-  // These events should only be bound after startup, since initialization above takes care of updating all contracts on startup
-  // Because there is now a event driven dependency, the lsp creator and lsp state updater must be in same process
-  serviceEvents.on("multiLspCreator", (event, data) => {
-    // handle created events
-    if (event === "created") {
-      console.log("LspCreator found a new contract", JSON.stringify(data));
-      services.lsps.updateLsps([data.address], data.startBlock, data.endBlock).catch(console.error);
+  new Promise((res) => setTimeout(res, updateRateS * 1000)).then(initLoops);
+
+  async function initLoops() {
+    console.log("Starting LSP API update loops");
+
+    // listen for new lsp contracts since after we have started api, and make sure they get state updated asap
+    // These events should only be bound after startup, since initialization above takes care of updating all contracts on startup
+    // Because there is now a event driven dependency, the lsp creator and lsp state updater must be in same process
+    serviceEvents.on("multiLspCreator", (event, data) => {
```

Comment:
One broader comment: it seems like we have a lot of top-level app files that are mostly repeated code with minor tweaks -- would it be preferable to just make them the same file with more configuration options to turn on and off certain features?

Also, what's the use case for an LSP-specific api?